source/code-snippets/usage-examples/findOne.ts

```diff
@@ -0,0 +1,42 @@
+import { MongoClient } from "mongodb";
+
+// Replace the uri string with your MongoDB deployment's connection string.
+const uri =
+  "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
+
+const client = new MongoClient(uri);
+
+interface IMDB {
+  rating: number;
+  votes: number;
+  id: number;
+}
+
+export interface Movie {
+  title: string;
+  year: number;
+  released: Date;
+  plot: string;
+  type: 'movie' | 'series';
+  imdb: IMDB;
+}
+
+async function run(): Promise<void> {
+  try {
+    await client.connect();
+
+    const database = client.db("sample_mflix");
+    // Specifying a Schema is always optional, but it enables type hinting on
+    // finds and inserts
+    const movies = database.collection<Movie>("movies");
+
+    const movie = await movies.findOne({ title: "The Room" }, {
+      sort: { rating: -1 },
+      projection: { _id: 0, title: 1, imdb: 1 },
```

Comment:
Using a projection and getting accurate type information requires you to define the resulting type of your query. So we could simplify this example by dropping the projection or we can describe some expected/best practice here. 

Projections are often used to make a query directly return an edited document that's acceptable to send out over a REST api etc. So this is what someone could do:

```typescript
interface MovieSummary {
  title: string;
  imdb: IMDB;
}
// or 
type MovieSummary = Pick<Movie, 'title' | 'imdb'>

// ...

const movieSummary = await movies.findOne<MovieSummary>({ title: "The Room" }, {
  sort: { rating: -1 },
  projection: { _id: 0, title: 1, imdb: 1 },
}); // Returns FindCursor<MovieSummary>
```

The `Pick<>` is one of Typescript's utility types: https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys Pick and Omit lend themselves well to quickly defining an accurate projection type.

Reply:
Oh nice! That seems super helpful for projections. Adding the suggested change now using `Pick`.